File: src/test/java/graphql/kickstart/tools/util/PrimitivesTest.java
package graphql.kickstart.tools.util;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

public class PrimitivesTest {
    @Test
    public void testWrapPrimitive() {
        Class<?> clazz = Primitives.wrap(int.class);

        assertEquals(Integer.class, clazz);
    }

    @Test
    public void testWrapNonPrimitive() {
        Class<?> clazz = Primitives.wrap(String.class);

        assertEquals(String.class, clazz);
    }

    @Test
    public void testWrapNull() {
        Class<?> clazz = Primitives.wrap(null);

        assertNull(clazz);
    }
}

File: src/test/kotlin/graphql/kickstart/tools/util/PrimitivesTest.kt
package graphql.kickstart.tools.util

import org.junit.Assert
import org.junit.Test

class PrimitivesTest {

    @Test
    fun `should wrap primitive Java class`() {
        val clazz = Primitives.wrap(Int::class.javaPrimitiveType)
        Assert.assertEquals(Integer::class.java, clazz)
    }

    @Test
    fun `should not wrap non-primitive class`() {
        val clazz = Primitives.wrap(String::class.java)
        Assert.assertEquals(String::class.java, clazz)
    }

    @Test
    fun `should return null for null`() {
        val clazz = Primitives.wrap<Any>(null)
        Assert.assertNull(clazz)
    }
}
